src/ledger/backend/readconn.ts

```diff
@@ -12,15 +15,20 @@ export const ErrErdstallContractNotConnected = new Error(
 	"erdstall contract not connected",
 );
 
-export interface LedgerReader extends ErdstallWatcher, Contracter {}
+export interface LedgerReader
+	extends NFTMetadataProvider,
+		ErdstallWatcher,
+		Contracter {}
 
 export class LedgerReadConn implements LedgerReader {
 	readonly contract: Erdstall;
 	private eventCache: Map<Function, (args: Array<any>) => void>;
+	private metadataUriCache: Map<string, NFTMetadata>;
 
 	constructor(contract: Erdstall) {
 		this.contract = contract;
 		this.eventCache = new Map<Function, (args: Array<any>) => void>();
+		this.metadataUriCache = new Map<string, NFTMetadata>();
```

Comment:
you can set those two directly where you define the fields, then you don't need to set them in the ctor.

Reply:
Any advantage you can point out? The way I always view it is that the member definitions (unitialised) describe the shape while the constructor always initializes (paints the picture :art:).

One could even go further and completely skip the member definition for passed params and define and default assign them in the constructor arguments. There is so many ways and I **think** we currently mainly first define and then assign them separately.

Reply:
But I can shorten the constructors, because the types can be inferred. Quite old code right here.